scalahost/nsc/src/main/scala/scala/meta/internal/semantic/InputOps.scala

```diff
@@ -19,7 +19,14 @@ trait InputOps { self: DatabaseOps =>
     def toMeta: m.Position = {
       assert(pos.isRange)
       val input = m.Input.File(pos.source.toAbsolutePath)
-      m.Position.Range(input, m.Point.Offset(input, pos.start), m.Point.Offset(input, pos.end))
+      val mstart = m.Point.Offset(input, pos.start)
+      val mend = m.Point.Offset(input, pos.end)
+      if (pos.point > pos.start && pos.point < pos.end) {
```

Comment:
being paranoid here, but can `pos.point` be `< pos.start` or `> pos.end` under any circumstance?
Would `if (point.point != pos.start && pos.point != pos.end)` be more defensive towards compiler's quirks?